rnd/autogpt_server/autogpt_server/data/queue.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+import os
+from abc import ABC, abstractmethod
+from datetime import datetime
+
+from redis.asyncio import Redis
+
+from autogpt_server.data.execution import ExecutionResult
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class DateTimeEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, datetime):
+            return o.isoformat()
+        return super().default(o)
+
+
+class AsyncEventQueue(ABC):
+    @abstractmethod
+    async def connect(self):
+        pass
+
+    @abstractmethod
+    async def close(self):
+        pass
+
+    @abstractmethod
+    async def put(self, execution_result: ExecutionResult):
+        pass
+
+    @abstractmethod
+    async def get(self) -> ExecutionResult | None:
+        pass
+
+
+class AsyncRedisEventQueue(AsyncEventQueue):
+
+    def __init__(self):
+        self.host = os.getenv("REDIS_HOST", "localhost")
+        self.port = int(os.getenv("REDIS_PORT", "6379"))
+        self.password = os.getenv("REDIS_PASSWORD", None)
+        self.queue_name = os.getenv("REDIS_QUEUE", "execution_events")
+        self.connection = None
+
+    async def connect(self):
+        if not self.connection:
+            self.connection = Redis(
+                host=self.host,
+                port=self.port,
+                password=self.password,
+                decode_responses=True,
+            )
+            await self.connection.ping()
+            logging.info(f"Connected to Redis on {self.host}:{self.port}")
```

Comment:
```suggestion
            logger.info(f"Connected to Redis on {self.host}:{self.port}")
```
same for the other occurrences in this file